src/coreclr/tools/Common/TypeSystem/Common/TypeSystemHelpers.cs

```diff
@@ -296,9 +296,16 @@ public static MethodDesc ResolveInterfaceMethodTarget(this TypeDesc thisType, Me
             }
             while (result == null && currentType != null);
 
+            // If no base type contains interface method, attempt to find default implementation
+            if (result == null)
+            {
+                _ = thisType.ResolveInterfaceMethodToDefaultImplementationOnType(interfaceMethodToResolve, out result);
+            }
+
```

Comment:
ILVerify needs to call `ResolveInterfaceMethodToDefaultImplementationOnType` manually after this resolution fails; it cannot be done in this method. There are nuances around default interface method resolution that will not work as expected if this is done here. The nuances might not affect ILVerify right now, but will affect it eventually.

E.g. if we want to do reporting of diamond cases in ILVerify (when multiple interfaces on the type provide a default implementation for a method), we'll have to call into ResolveInterfaceMethodToDefaultImplementationOnType manually anyway.

Reply:
(This change breaks other consumers of TypeSystemHelper.cs)

Reply:
Could you elaborate on what you mean with `ILVerify needs to call ResolveInterfaceMethodToDefaultImplementationOnType manually after this resolution fails`? Why is it a problem that ILVerify needs to call the method after the resolution fails? Does the method affect state in some way, meaning a second call won't succeed anymore?

Reply:
`ResolveInterfaceMethodTarget` (which is in a source file shared with other components in this repo and not unique to ILVerify) should not call `ResolveInterfaceMethodToDefaultImplementationOnType`. Instead ILVerify should call `ResolveInterfaceMethodToDefaultImplementationOnType` when `ResolveInterfaceMethodTarget` returns null.

This is how other components do this as well. We don't want to hide failure modes that are unique to default interface methods here (I'm referring to the value returned from `ResolveInterfaceMethodToDefaultImplementationOnType` that is being swallowed here).

Reply:
Right I think I understand correctly now. So would it be sufficient to move the call to `ResolveInterfaceMethodToDefaultImplementationOnType` into `VerifyInterfaces` if  `ResolveInterfaceMethodToDefaultImplementationOnType` returns `null`?

Reply:
Yes, that sound good. It will also put ILVerify into a better position to report unique error messages for things like the diamond case.

Reply:
Thank you! I adjusted the callsite of `ResolveInterfaceMethodToDefaultImplementationOnType`. Apologies for misunderstanding your original comment, thank you for elaborating :)
I'll create another issue for the bug mentioned in this PR's description.